src/lib.rs

```diff
@@ -491,3 +499,14 @@ enum DerInner<'a> {
     Owned(Vec<u8>),
     Borrowed(&'a [u8]),
 }
+
+#[cfg(feature = "alloc")]
+impl DerInner<'_> {
+    fn into_owned(self) -> DerInner<'static> {
+        let vec = match self {
+            Self::Owned(vec) => vec,
+            Self::Borrowed(slice) => slice.to_vec(),
+        };
+        DerInner::Owned(vec)
```

Comment:
I think this is a little bit cleaner all-in-one:

```suggestion
        DerInner::Owned(match self {
            Self::Owned(vec) => vec,
            Self::Borrowed(slice) => slice.to_vec(),
        })
```